sklearn/utils/_testing.py

```diff
@@ -1060,3 +1060,32 @@ def transform(self, X, y=None):
 
     def fit_transform(self, X, y=None):
         return self.fit(X, y).transform(X, y)
+
+
+def generate_dataframe_from_lib(dataframe_lib, data):
```

Comment:
Can we move this block next to the `_convert_container` function?
I am wondering if you think that those 2 functions should be kind of merge or it is better to keep them separated.